odl/space/fspace.py

```diff
@@ -1323,8 +1330,11 @@ def __call__(self, x, out=None, **kwargs):
 
                 if out_shape != () and out.shape != out_shape:
                     # Broadcast the returned element, but not in the
-                    # scalar case.
+                    # scalar case. The resulting array may be read-only,
+                    # in which case we copy.
                     out = broadcast_to(out, out_shape)
+                    if not out.flags.writeable:
```

Comment:
Are there tests in place for corner cases like this?

Reply:
Now yes :-)
This is not a corner case, it applies whenever broadcasting happens.